tests/e2e/sanitizer_test.go

```diff
@@ -0,0 +1,400 @@
+	deletedElements, err := fileDivElement.FindElements(selenium.ByTagName, "del")
+
+	if err != nil {
+		t.Fatal("Could not find deleted elements", err)
+	}
+	insertedElements, err := fileDivElement.FindElements(selenium.ByTagName, "ins")
+
+	if err != nil {
+		t.Fatal("Could not find inserted elements", err)
+	}
+
+	lineNumElements, err := fileDivElement.FindElements(selenium.ByCSSSelector, ".d2h-code-linenumber.d2h-del.d2h-change")
+
+	if err != nil {
+		t.Fatal("Could not find line numbers", err)
+	}
+
+	assert.Equal(t, len(deletedElements), len(insertedElements), "Deleted Elements length is different from inserted elements: %d vs. %d", len(deletedElements), len(insertedElements))
+
+	for index, deletedElement := range deletedElements {
+		t.Logf("Deleted Elements: %d\nInserted Elements: %d\n", len(deletedElements), len(insertedElements))
+		deletedElementText, err := deletedElement.Text()
+
+		if err != nil {
+			return err
+		}
+
+		insertedElementText, err := insertedElements[index].Text()
+
+		if err != nil {
+			return err
+		}
+		expectedDeletedValue := expectedValues[index][0]
+		expectedInsertedValue := expectedValues[index][1]
+
+		t.Logf("Deleted Elements: %s\nInserted Elements: %s\n", deletedElementText, insertedElementText)
+		assert.Equal(t, expectedDeletedValue, deletedElementText, "Did not delete expected parameter (%s), but removed {%s}", expectedDeletedValue, deletedElementText)
+
+		if expectedInsertedValue == contextualReplacement {
+			mappedInsertedValue, ok := contextualReplacementMapping[expectedDeletedValue]
+			if !ok {
+				contextualReplacementMapping[expectedDeletedValue] = insertedElementText
+			} else {
+				assert.Equal(t, mappedInsertedValue, insertedElementText, "Contextual Replacement shown different values. Expected %s, but got %s", mappedInsertedValue, insertedElementText)
+			}
+			expectedInsertedValue = insertedElementText
+
+		} else {
+			assert.Equal(t, expectedInsertedValue, insertedElementText, "Expected inserted value is %s, but got %s", expectedInsertedValue, insertedElementText)
+		}
+
+		lineNum1Element, err := lineNumElements[index].FindElement(selenium.ByCSSSelector, ".line-num1")
+		if err != nil {
+			return err
+		}
+
+		lineNum1Value, err := lineNum1Element.Text()
+
+		if err != nil {
+			return err
+		}
+
+		lineNum, err := strconv.Atoi(lineNum1Value)
+
+		if err != nil {
+			return fmt.Errorf("could not convert number %s: %s", lineNum1Value, err)
+		}
+
+		sanitizedFileChangeLines[lineNum] = expectedInsertedValue
+	}
+	return nil
+}
+
+func validE2EProcess(t *testing.T, webDriver selenium.WebDriver) error {
+	var err error
+	t.Log("Starting validating e2e process...\n")
+	filesToSanitize := make([]string, len(sanitizeFileToExpectedSanitation)+len(untouchedFileNames))
+
+	i := 0
+	for fileToSanitize := range sanitizeFileToExpectedSanitation {
+		filesToSanitize[i] = fileToSanitize
+		i++
+	}
+	for index := range untouchedFileNames {
+		filesToSanitize[i] = untouchedFileNames[index]
+		i++
```

Comment:
You can just append to the slice instead of using an index.
https://go.dev/tour/moretypes/15